docs/en/v0.4/user-guide/upgrade.md

```diff
@@ -0,0 +1,128 @@
+# Upgrade
+
+From `v0.4`, we provide a built-in utility to help upgrade your previous GreptimeDB deployment to the latest version.
+It's recommended to use this method to upgrade your GreptimeDB with different versions.
+
+The lowest version that can be upgraded by this tool is `v0.3.0`.
+
+## CLI
+
+This tool is in the `greptime` binary. You need to prepare the corresponding binary of target version before start.
+
+```shell
+greptime cli export --help
+```
+
+And the help text is as follows:
+
+```shell
+greptime-cli-export 
+
+USAGE:
+    greptime cli export [OPTIONS] --addr <ADDR> --output-dir <OUTPUT_DIR> --target <TARGET>
+
+OPTIONS:
+        --addr <ADDR>                  Server address to connect
+        --database <DATABASE>          The name of the catalog to export. Default to "greptime-*""
+                                       [default: ]
+    -h, --help                         Print help information
+    -j, --export-jobs <EXPORT_JOBS>    Parallelism of the export [default: 1]
+        --max-retry <MAX_RETRY>        Max retry times for each job [default: 3]
+        --output-dir <OUTPUT_DIR>      Directory to put the exported data. E.g.:
+                                       /tmp/greptimedb-export
+    -t, --target <TARGET>              Things to export [possible values: create-table, table-data]
+```
+
+Here explains the meaning of some important options
+
+- `--addr`: The address of the Frontend node.
+- `--output-dir`: The directory to put the exported data. Give a path at your current machine. The exported SQL files will be put in that directory.
+- `--target`: The things to export. `create-table` can export the `CREATE TABLE` clause for each table. `table-data` can export the data of each table alone with `COPY FROM` clause.
+
+For a complete upgrade, you will need to execute this tools twice with each target options.
+
+## Example
+
+Here is a complete example for upgrading from `v0.3.0` to `v0.4.0`.
+
+In the following text, we assume that you have a Frontend's gRPC endpoint is available at `127.0.0.1:4000`. And the catalog is `greptime`, schema is `public`. Hence the database is `greptime-public`. The output dir is `/tmp/greptimedb-export`.
```

Comment:
Should we mention `catalog=greptime` here?

Reply:
I find the entire `--database` isn't necessary, the default value `greptime-*` is good enough...